rq/worker.py

```diff
@@ -257,6 +257,12 @@ def register_death(self):
             p.expire(self.key, 60)
             p.execute()
 
+    def set_warm_shutdown_requested_date(self):
+        """Sets the date on which the worker received a warm shutdown request"""
+        with self.connection._pipeline() as p:
```

Comment:
I don't think pipeline is needed as we're only updating one key. If this method is not used in more than one places, perhaps we could just inline the code in "request_stop", since it's a one liner anyway.